## 05 Merge Interim Data of all Players (FBREF and Transfermarkt)

In [1]:
import pandas as pd
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

In [2]:
import sys
sys.path.append("..")
from src.preprocessing.merge_player_data import run_merge_pipeline

merge_result: tuple = run_merge_pipeline("Valencia CF", ["2223", "2324", "2425"])
df_all_players = merge_result[0] if isinstance(merge_result, tuple) else merge_result

print(f"Merge completed successfully!")
print(f"Total merged rows: {df_all_players.shape[0]}")
print(f"Total columns: {df_all_players.shape[1]}")

Attempting fuzzy matching for 51 unmatched players...
Fuzzy match: 'nicolás gonzález' -> 'nico gonzález' (score: 92)
Fuzzy match: 'javier guerra' -> 'javi guerra' (score: 92)
Fuzzy match: 'jesus vazquez' -> 'jesús vázquez' (score: 92)
Fuzzy match: 'emilio bernard' -> 'emilio bernad' (score: 96)
Fuzzy match: 'charlie perez' -> 'charlie pérez' (score: 96)
Fuzzy match: 'javier guerra' -> 'javi guerra' (score: 92)
Fuzzy match: 'jesus vazquez' -> 'jesús vázquez' (score: 92)
Fuzzy match: 'pablo gozalbez' -> 'pablo gozálbez' (score: 96)
Fuzzy match: 'marco camús' -> 'marco camus' (score: 95)
Fuzzy match: 'javier guerra' -> 'javi guerra' (score: 92)
Fuzzy match: 'jesus vazquez' -> 'jesús vázquez' (score: 92)
Dropped Nation_* columns: ['Nation_player_shooting', 'Nation_player_passing', 'Nation_player_passing_types', 'Nation_player_gca', 'Nation_player_defense', 'Nation_player_possession']
Dropped Age columns: ['Age_player_shooting', 'Age_player_passing', 'Age_player_passing_types', 'Age_player_

In [3]:
total_with_market_value = df_all_players['MarketValueEuro'].notna().sum()
print(f"Rows with market value data: {total_with_market_value}/{len(df_all_players)} ({total_with_market_value/len(df_all_players)*100:.1f}%)")

Rows with market value data: 80/114 (70.2%)


In [4]:
print("\nSample merged data:")
sample_cols = ['Player', 'Season', 'MarketValueEuro', 'Age_x', 'Position']
available_cols = [col for col in sample_cols if col in df_all_players.columns]
display(df_all_players[available_cols].head(20))


Sample merged data:


Player Season  MarketValueEuro          Position
21         Alberto Marí   2223         800000.0    Centre-Forward
43         Alberto Marí   2324        1500000.0    Centre-Forward
109        Alberto Marí   2425              NaN               NaN
18         Carlos Soler   2223       25000000.0  Central Midfield
12         Cenk Özkacar   2223        5000000.0       Centre-Back
35         Cenk Özkacar   2324        4000000.0       Centre-Back
117        Cenk Özkacar   2425              NaN               NaN
78        Charlie Perez   2223          25000.0        Goalkeeper
19   Cristhian Mosquera   2223        1000000.0       Centre-Back
31   Cristhian Mosquera   2324       30000000.0       Centre-Back
50   Cristhian Mosquera   2425       30000000.0       Centre-Back
76             Cristian   2223              NaN               NaN
90             Cristian   2324              NaN               NaN
47        César Tárrega   2324        1500000.0       Centre-Back
53        César Tárrega   2425       10000000.0       Centre-Back
101          Dani Gómez   2425              NaN               NaN
48         David Otorbi   2324         500000.0      Right Winger
108        David Otorbi   2425              NaN               NaN
16          Diego López   2223        1500000.0      Right Winger
32          Diego López   2324       14000000.0      Right Winger

In [5]:
# Group by player and sort by season to see all seasons for each player with missing market values
players_with_missing_market_values = df_all_players[df_all_players['MarketValueEuro'].isna()].sort_values(['Player', 'Season'])
display(players_with_missing_market_values[['Player', 'Season', 'Age', 'Position']].head(40))

Player Season  Age Position
109                    Alberto Marí   2425   23      NaN
117                    Cenk Özkacar   2425   23      NaN
76                         Cristian   2223   24      NaN
90                         Cristian   2324   25      NaN
101                      Dani Gómez   2425   26      NaN
108                    David Otorbi   2425   16      NaN
70   Domingos André Ribeiro Almeida   2223   22      NaN
86   Domingos André Ribeiro Almeida   2324   23      NaN
97   Domingos André Ribeiro Almeida   2425   24      NaN
74                  Francisco Perez   2223   19      NaN
84                  Francisco Perez   2324   20      NaN
102                 Francisco Perez   2425   21      NaN
104                   Germán Valera   2425   22      NaN
112                   Hamza Bellari   2425   21      NaN
107                    Iker Córdoba   2425   18      NaN
103              Ivan Jaime Pajuelo   2425   23      NaN
79                   Jesús Santiago   2223   18      NaN
82            José Luis García Vayá   2324   24      NaN
99            José Luis García Vayá   2425   25      NaN
69                   José Luis Gayà   2223   27      NaN
85                   José Luis Gayà   2324   28      NaN
98                   José Luis Gayà   2425   29      NaN
116                     Lucas Núñez   2425   18      NaN
105            Maximiliano Caufriez   2425   27      NaN
111                    Pedro Alemañ   2425   22      NaN
88                   Peter González   2324   21      NaN
113                    Raúl Jiménez   2425   18      NaN
106                  Rodrigo Abajas   2425   21      NaN
77                   Ruben Lendinez   2223   19      NaN
91                   Ruben Lendinez   2324   20      NaN
114                  Ruben Lendinez   2425   21      NaN
92                Vicent Abril Sanz   2324   18      NaN
110               Vicent Abril Sanz   2425   19      NaN
115                 Warren Madrigal   2425   20      NaN

In [6]:
# Remove system/total rows
df_clean = df_all_players[~df_all_players['Player'].str.contains('Total|Squad|Opponent', case=False)]

# Only players with market values
df_with_market_values = df_all_players[df_all_players['MarketValueEuro'].notna()]

In [7]:
df_with_market_values

Player Season   Nation                   Nationality  \
21           Alberto Marí   2223   es ESP                         Spain   
43           Alberto Marí   2324   es ESP                         Spain   
18           Carlos Soler   2223   es ESP                         Spain   
12           Cenk Özkacar   2223   tr TUR                       Türkiye   
35           Cenk Özkacar   2324   tr TUR                       Türkiye   
78          Charlie Perez   2223   es ESP                         Spain   
19     Cristhian Mosquera   2223   es ESP         ['Spain', 'Colombia']   
31     Cristhian Mosquera   2324   es ESP         ['Spain', 'Colombia']   
50     Cristhian Mosquera   2425   es ESP         ['Spain', 'Colombia']   
47          César Tárrega   2324   es ESP                         Spain   
53          César Tárrega   2425   es ESP                         Spain   
48           David Otorbi   2324   es ESP          ['Spain', 'Nigeria']   
16            Diego López   2223   es ESP                         Spain   
32            Diego López   2324   es ESP                         Spain   
51            Diego López   2425   es ESP                         Spain   
9       Dimitri Foulquier   2223   fr FRA      ['Guadeloupe', 'France']   
34      Dimitri Foulquier   2324   fr FRA      ['Guadeloupe', 'France']   
56      Dimitri Foulquier   2425   fr FRA      ['Guadeloupe', 'France']   
5          Edinson Cavani   2223   uy URU          ['Uruguay', 'Italy']   
75         Emilio Bernard   2223   es ESP                         Spain   
55       Enzo Barrenechea   2425   ar ARG                     Argentina   
6             Eray Cömert   2223   ch SUI    ['Switzerland', 'Türkiye']   
8        Gabriel Paulista   2223   br BRA           ['Brazil', 'Spain']   
36       Gabriel Paulista   2324   br BRA           ['Brazil', 'Spain']   
0    Giorgi Mamardashvili   2223   ge GEO                       Georgia   
29   Giorgi Mamardashvili   2324   ge GEO                       Georgia   
52   Giorgi Mamardashvili   2425   ge GEO                       Georgia   
13              Hugo Duro   2223   es ESP                         Spain   
30              Hugo Duro   2324   es ESP                         Spain   
57              Hugo Duro   2425   es ESP                         Spain   
45          Hugo González   2324   es ESP                         Spain   
7          Hugo Guillamón   2223   es ESP                         Spain   
38         Hugo Guillamón   2324   es ESP                         Spain   
63         Hugo Guillamón   2425   es ESP                         Spain   
24          Iago Herrerín   2223   es ESP                         Spain   
14           Ilaix Moriba   2223   gn GUI           ['Guinea', 'Spain']   
22         Jaume Doménech   2223   es ESP                         Spain   
44         Jaume Doménech   2324   es ESP                         Spain   
68         Jaume Doménech   2425   es ESP                         Spain   
72          Javier Guerra   2223   es ESP                         Spain   
83          Javier Guerra   2324   es ESP                         Spain   
96          Javier Guerra   2425   es ESP                         Spain   
73          Jesus Vazquez   2223   es ESP                         Spain   
87          Jesus Vazquez   2324   es ESP                         Spain   
100         Jesus Vazquez   2425   es ESP                         Spain   
11        Justin Kluivert   2223   nl NED   ['Netherlands', 'Suriname']   
25         Koba Koindredi   2223   fr FRA   ['France', 'Neukaledonien']   
54             Luis Rioja   2425   es ESP                         Spain   
28           Manu Vallejo   2223   es ESP                         Spain   
93            Marco Camús   2324   es ESP                         Spain   
17           Marcos André   2223   br BRA           ['Brazil', 'Spain']   
23        Mario Dominguez   2223   es ESP                         Spain   
46        Mario Dominguez   2324   es ESP   

NOTE: Current issues: 
- Nation and Nationality contain different values -----> Let's use 'Position' it's cleaner
- Pos and Position are not the same and we have in Pos sometimes multiple values -----> By keeping only last 3 characters in 'Pos' the data is cleaner and compact
- We have new players with only one season or missing values -----> Let's skip them then, it's best to look at players with 2-3+ seasons

In [8]:
#### Let's use this column instead of 'Pos' since it's cleaner
df_with_market_values['Position'].value_counts()

Position
Centre-Back           15
Centre-Forward        13
Goalkeeper            10
Central Midfield       9
Right-Back             7
Right Winger           6
Defensive Midfield     5
Attacking Midfield     5
Left Winger            5
Left-Back              4
Second Striker         1
Name: count, dtype: int64

In [9]:
df_with_market_values['Nation'].value_counts().head(3)

Nation
es ESP    47
fr FRA     4
br BRA     4
Name: count, dtype: int64

In [10]:
df_with_market_values['Nation'] = df_with_market_values['Nation'].str[-3:]

/var/folders/1h/14p077t95lbcy6x5tc6bqx3h0000gn/T/ipykernel_53065/3401751002.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_with_market_values['Nation'] = df_with_market_values['Nation'].str[-3:]


In [11]:
print(df_with_market_values.shape)
df_with_market_values.head(10)

(80, 122)


Player Season Nation            Nationality    Pos  \
21        Alberto Marí   2223    ESP                  Spain  FW,MF   
43        Alberto Marí   2324    ESP                  Spain     FW   
18        Carlos Soler   2223    ESP                  Spain     MF   
12        Cenk Özkacar   2223    TUR                Türkiye     DF   
35        Cenk Özkacar   2324    TUR                Türkiye     DF   
78       Charlie Perez   2223    ESP                  Spain     GK   
19  Cristhian Mosquera   2223    ESP  ['Spain', 'Colombia']     DF   
31  Cristhian Mosquera   2324    ESP  ['Spain', 'Colombia']     DF   
50  Cristhian Mosquera   2425    ESP  ['Spain', 'Colombia']     DF   
47       César Tárrega   2324    ESP                  Spain     DF   

            Position  Age  MP  Starts     Min   90s  Gls  Ast  G+A  G-PK   PK  \
21    Centre-Forward   21   5       0    86.0   1.0  1.0  0.0  1.0   1.0  0.0   
43    Centre-Forward   22  16       2   276.0   3.1  1.0  1.0  2.0   0.0  1.0   
18  Central Midfield   25   3       3   242.0   2.7  1.0  0.0  1.0   0.0  1.0   
12       Centre-Back   21  17      14  1365.0  15.2  0.0  0.0  0.0   0.0  0.0   
35       Centre-Back   22  23      18  1598.0  17.8  0.0  0.0  0.0   0.0  0.0   
78        Goalkeeper   20   0       0     NaN   NaN  NaN  NaN  NaN   NaN  NaN   
19       Centre-Back   18   3       1   145.0   1.6  0.0  0.0  0.0   0.0  0.0   
31       Centre-Back   19  36      33  3075.0  34.2  0.0  0.0  0.0   0.0  0.0   
50       Centre-Back   20  37      37  3319.0  36.9  1.0  0.0  1.0   1.0  0.0   
47       Centre-Back   21   1       0     8.0   0.1  0.0  0.0  0.0   0.0  0.0   

    PKatt  CrdY  CrdR   xG  npxG  xAG  npxG+xAG  PrgC   PrgP  PrgR  Gls.1  \
21    0.0   1.0   0.0  0.2   0.2  0.0       0.2   1.0    1.0   6.0   1.05   
43    1.0   0.0   0.0  1.8   1.2  0.9       2.0   2.0    6.0  13.0   0.33   
18    1.0   0.0   0.0  0.9   0.2  0.2       0.4   3.0   13.0   7.0   0.37   
12    0.0   1.0   0.0  0.8   0.8  0.2       1.0  23.0   61.0   2.0   0.00   
35    0.0   3.0   0.0  0.3   0.3  0.1       0.3  17.0   56.0   5.0   0.00   
78    NaN   NaN   NaN  NaN   NaN  NaN       NaN   NaN    NaN   NaN    NaN   
19    0.0   0.0   0.0  0.2   0.2  0.0       0.2   0.0    1.0   1.0   0.00   
31    0.0   2.0   0.0  0.7   0.7  0.0       0.7  14.0   77.0   4.0   0.00   
50    0.0   6.0   0.0  0.4   0.4  0.0       0.4  35.0  123.0   6.0   0.03   
47    0.0   0.0   0.0  0.0   0.0  0.0       0.0   0.0    0.0   0.0   0.00   

    Ast.1  G+A.1  G-PK.1  G+A-PK  xG.1  xAG.1  xG+xAG  npxG.1  npxG+xAG.1  \
21   0.00   1.05    1.05    1.05  0.25   0.00    0.25    0.25        0.25   
43   0.33   0.65    0.00    0.33  0.60   0.28    0.88    0.39        0.67   
18   0.00   0.37    0.00    0.00  0.35   0.08    0.43    0.06        0.14   
12   0.00   0.00    0.00    0.00  0.05   0.02    0.07    0.05        0.07   
35   0.00   0.00    0.00    0.00  0.01   0.00    0.02    0.01        0.02   
78    NaN    NaN     NaN     NaN   NaN    NaN     NaN     NaN         NaN   
19   0.00   0.00    0.00    0.00  0.12   0.00    0.12    0.12        0.12   
31   0.00   0.00    0.00    0.00  0.02   0.00    0.02    0.02        0.02   
50   0.00   0.03    0.03    0.03  0.01   0.00    0.01    0.01        0.01   
47   0.00   0.00    0.00    0.00  0.00   0.00    0.00    0.00        0.00   

      Sh  SoT  SoT%  Sh/90  SoT/90  G/Sh  G/SoT  Dist   FK  npxG/Sh  G-xG  \
21   2.0  1.0  50.0   2.09    1.05   0.5    1.0  10.7  0.0     0.12   0.8   
43   6.0  2.0  33.3   1.96    0.65   0.0    0.0  11.4  0.0     0.20  -0.8   
18   4.0  1.0  25.0   1.49    0.37   0.0    0.0  30.0  3.0     0.04   0.1   
12  11.0  2.0  18.2   0.73    0.13   0.0    0.0  12.8  0.0     0.07  -0.8   
35   6.0  3.0  50.0   0.34    0.17   0.0    0.0  14.3  0.0     0.04  -0.3   
78   NaN  NaN   NaN    NaN     NaN   NaN    NaN   NaN  NaN      NaN   NaN   
19   2.0  0.0   0.0   1.24    0.00   0.0    NaN   9.3  0.0     0.10  -0.2   
31   9.0  2.0  22.2   0.26    0.06   0.0 

In [12]:
df_with_market_values['Nation'].value_counts().head(3)

Nation
ESP    47
FRA     4
BRA     4
Name: count, dtype: int64

In [13]:
from sklearn.preprocessing import LabelEncoder

le = LabelEncoder()

df_with_market_values['Nation'] = le.fit_transform(df_with_market_values['Nation'])
df_with_market_values['Position'] = le.fit_transform(df_with_market_values['Position'])

/var/folders/1h/14p077t95lbcy6x5tc6bqx3h0000gn/T/ipykernel_53065/1260227516.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_with_market_values['Nation'] = le.fit_transform(df_with_market_values['Nation'])
/var/folders/1h/14p077t95lbcy6x5tc6bqx3h0000gn/T/ipykernel_53065/1260227516.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_with_market_values['Position'] = le.fit_transform(df_with_market_values['Position'])


In [22]:
correlation_with_market_value = df_with_market_values.corrwith(df_with_market_values['MarketValueEuro'])

print(correlation_with_market_value.sort_values(ascending=False).head(11))

MarketValueEuro    1.000000
Starts             0.718316
90s                0.693411
Min                0.693269
Cmp.3              0.645374
PrgDist            0.641521
Att.3              0.638499
Def_Pen            0.621067
MP                 0.609923
TotDist            0.591553
Carries            0.568287
dtype: float64


### 🔟 Top 10 Features Most Correlated with Market Value

- **Starts (0.718)** — Indicates how often a player is trusted to start matches; more starts = more value.
- **90s (0.693)** — Reflects total playing time in 90-minute equivalents; regular play boosts visibility and importance.
- **Min (0.693)** — Total minutes played; more minutes typically means more experience and trust from coaches.
- **Cmp.3 (0.645)** — Completed passes in the attacking third; shows involvement in crucial offensive phases.
- **PrgDist (0.642)** — Total progressive carrying distance; highlights ability to move the ball forward effectively.
- **Att.3 (0.638)** — Passes attempted in the attacking third; shows a player’s role in build-up and attacking moves.
- **Def_Pen (0.621)** — Defensive actions inside the penalty area; key for evaluating defenders’ last-line contributions.
- **MP (0.610)** — Matches played; simple yet powerful indicator of consistency and fitness.
- **TotDist (0.592)** — Total distance covered while carrying the ball; proxies physical activity and involvement.
- **Carries (0.568)** — Number of ball carries; emphasizes players who actively retain and progress possession.


- Nation and Position don't have an impact on Market Value

### Interrogations:
- When scraping data from other teams to build a regression model to predict a player's next year's market value, I believe we should consider the ranking of the team. 
- A player for Valencia might not be value the same as a Real Madrid player even with the same performance.
- That's because Real Madrid plays bigger competitions and is more likely to win titles which boosts the player's exposure and therefore the market price.
- number of competitions played as well

## Other spanish teams to scrape for ML:

- https://fbref.com/en/squads/53a2f082/2024-2025/Real-Madrid-Stats
- https://fbref.com/en/squads/206d90db/Barcelona-Stats
- https://fbref.com/en/squads/ad2be733/Sevilla-Stats
- https://fbref.com/en/squads/db3b9613/Atletico-Madrid-Stats
- https://fbref.com/en/squads/2b390eca/Athletic-Club-Stats
- https://fbref.com/en/squads/2a8183b3/Villarreal-Stats
- https://fbref.com/en/squads/e31d1cd9/Real-Sociedad-Stats
- https://fbref.com/en/squads/fc536746/Real-Betis-Stats

---

# Merge all player data

In [3]:
import sys
import os
from pathlib import Path
sys.path.append("..")
from src.preprocessing.merge_player_data import run_merge_pipeline

teams_list: list[str] = [
    "Real Madrid CF",
    "FC Barcelona", 
    "Atlético Madrid",
    "Sevilla FC", 
    "Athletic Club",
    "Villarreal CF",
    "Real Sociedad",
    "Real Betis",
    "Valencia CF",
]

seasons_list: list[str] = ["2223", "2324", "2425"]

for team_name in teams_list:
    print(f"Processing {team_name}...")
    
    # Create directory structure
    team_directory: Path = Path(f"../data/interim/{team_name}/merged")
    team_directory.mkdir(parents=True, exist_ok=True)
    
    # Run merge pipeline for current team
    merge_result: tuple = run_merge_pipeline(team_name, seasons_list)
    df_all_players = merge_result[0] if isinstance(merge_result, tuple) else merge_result
    
    print(f"Merge completed for {team_name}!")
    print(f"Total merged rows: {df_all_players.shape[0]}")
    print(f"Total columns: {df_all_players.shape[1]}")
    
    # Remove system/total rows
    player_column_filled = df_all_players['Player'].fillna('Unknown')
    df_clean = df_all_players[~player_column_filled.str.contains('Total|Squad|Opponent', case=False)]

    # Only players with market values
    df_with_market_values = df_all_players[df_all_players['MarketValueEuro'].notna()]
    
    # Clean nation column
    df_with_market_values['Nation'] = df_with_market_values['Nation'].str[-3:]
    
    # Save processed data
    clean_filename: str = f"{team_name.replace(' ', '_').lower()}_clean.csv"
    market_values_filename: str = f"{team_name.replace(' ', '_').lower()}_market_values.csv"
    
    df_clean.to_csv(team_directory / clean_filename, index=False)
    df_with_market_values.to_csv(team_directory / market_values_filename, index=False)
    
    print(f"Saved {clean_filename} and {market_values_filename} to {team_directory}")
    print("-" * 50)

print("All teams processed successfully!")


Processing Real Madrid CF...
Current working directory: /Users/markuskuehnle/Documents/projects/talent-value-forecast/notebooks
Found project root: /Users/markuskuehnle/Documents/projects/talent-value-forecast
Trying possible paths: ['/Users/markuskuehnle/Documents/projects/talent-value-forecast/data', '../data', '../../data', '/Users/markuskuehnle/Documents/projects/talent-value-forecast/data', '/Users/markuskuehnle/Documents/projects/data', 'data']
Checking path: /Users/markuskuehnle/Documents/projects/talent-value-forecast/data - exists: True
Found data directory: /Users/markuskuehnle/Documents/projects/talent-value-forecast/data
Looking for FBref directory: /Users/markuskuehnle/Documents/projects/talent-value-forecast/data/interim/Real Madrid CF/fbref
Looking for Transfermarkt file: /Users/markuskuehnle/Documents/projects/talent-value-forecast/data/interim/Real Madrid CF/transfermarkt/real_madrid_cf_2020_2024.csv
Attempting fuzzy matching for 41 unmatched players...
Fuzzy match: 'v

/var/folders/0z/26cwjfqx4fj73hndqg4mkgl80000gn/T/ipykernel_8628/1532477885.py:44: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_with_market_values['Nation'] = df_with_market_values['Nation'].str[-3:]
/var/folders/0z/26cwjfqx4fj73hndqg4mkgl80000gn/T/ipykernel_8628/1532477885.py:44: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_with_market_values['Nation'] = df_with_market_values['Nation'].str[-3:]
/var/folders/0z/26cwjfqx4fj73hndqg4mkgl80000gn/T/ipykernel_8628/1532477885.py:44: SettingWithCopyWarning: 

Attempting fuzzy matching for 59 unmatched players...
Fuzzy match: 'ivan rakitić' -> 'ivan rakitic' (score: 92)
Fuzzy match: 'loïc bade' -> 'loïc badé' (score: 93)
Fuzzy match: 'marko dmitrović' -> 'marko dmitrovic' (score: 93)
Fuzzy match: 'loïc bade' -> 'loïc badé' (score: 93)
Fuzzy match: 'ivan rakitić' -> 'ivan rakitic' (score: 92)
Fuzzy match: 'marko dmitrović' -> 'marko dmitrovic' (score: 93)
Fuzzy match: 'dodi lukebakio' -> 'dodi lukébakio' (score: 96)
Fuzzy match: 'lucien agoume' -> 'lucien agoumé' (score: 96)
Fuzzy match: 'alejo véliz' -> 'alejo veliz' (score: 95)
Fuzzy match: 'dodi lukebakio' -> 'dodi lukébakio' (score: 96)
Fuzzy match: 'loïc bade' -> 'loïc badé' (score: 93)
Fuzzy match: 'lucien agoume' -> 'lucien agoumé' (score: 96)
Fuzzy match: 'ruben vargas' -> 'rubén vargas' (score: 96)
Dropped Nation_* columns: ['Nation_player_shooting', 'Nation_player_passing', 'Nation_player_passing_types', 'Nation_player_gca', 'Nation_player_defense', 'Nation_player_possession']
Dropp

/var/folders/0z/26cwjfqx4fj73hndqg4mkgl80000gn/T/ipykernel_8628/1532477885.py:44: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_with_market_values['Nation'] = df_with_market_values['Nation'].str[-3:]
/var/folders/0z/26cwjfqx4fj73hndqg4mkgl80000gn/T/ipykernel_8628/1532477885.py:44: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_with_market_values['Nation'] = df_with_market_values['Nation'].str[-3:]
/var/folders/0z/26cwjfqx4fj73hndqg4mkgl80000gn/T/ipykernel_8628/1532477885.py:44: SettingWithCopyWarning: 

Attempting fuzzy matching for 89 unmatched players...
Fuzzy match: 'mohamed ali cho' -> 'mohamed-ali cho' (score: 100)
Fuzzy match: 'alex sola' -> 'álex sola' (score: 94)
Fuzzy match: 'roberto navarro' -> 'robert navarro' (score: 97)
Fuzzy match: 'ander martin' -> 'ander martín' (score: 96)
Fuzzy match: 'jonathan gomez' -> 'jonathan gómez' (score: 96)
Dropped Nation_* columns: ['Nation_player_shooting', 'Nation_player_passing', 'Nation_player_passing_types', 'Nation_player_gca', 'Nation_player_defense', 'Nation_player_possession']
Dropped Age columns: ['Age_player_shooting', 'Age_player_passing', 'Age_player_passing_types', 'Age_player_gca', 'Age_player_defense', 'Age_player_possession', 'Age_y', 'Age']
Renamed 'Age_x' to 'Age'
Converted 'Age' to integer (NaN values set to -1)
Removed 43 duplicate columns with _player_* pattern
Removed columns: ['Player_player_shooting', 'Pos_player_shooting', '90s_player_shooting', 'Gls_player_shooting', 'PK_player_shooting', 'PKatt_player_shooting', 

/var/folders/0z/26cwjfqx4fj73hndqg4mkgl80000gn/T/ipykernel_8628/1532477885.py:44: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_with_market_values['Nation'] = df_with_market_values['Nation'].str[-3:]
